vicodyn/include/cocaine/vicodyn/peer.hpp

```diff
@@ -82,17 +83,26 @@ class peer_t : public std::enable_shared_from_this<peer_t> {
 // thread safe wrapper on map of peers indexed by uuid
 class peers_t {
 public:
-    class app_service_t {
-        using clock_t = std::chrono::steady_clock;
+    using clock_t = std::chrono::steady_clock;
 
+    class app_service_t {
         clock_t::time_point ban_until_;
+        std::unique_ptr<metrics::usts::ewma<clock_t>> timings_ewma_;
 
     public:
-        auto ban(const std::chrono::milliseconds& timeout) -> void;
+        app_service_t(clock_t::duration timings_window);
+
+        auto ban(std::chrono::milliseconds timeout) -> void;
 
         auto banned() const -> bool;
+
+        auto add_request_timing(clock_t::time_point start, clock_t::duration elapsed) -> void;
+
+        auto average_elapsed() const -> clock_t::duration;
     };
 
+    using app_service_f = std::function<bool(const app_service_t& app_service)>;
+    using peer_f = std::function<bool(const peer_t& peer_service)>;
```

Comment:
It is usually plugin-wide naming conventions you could follow.

Reply:
What is it mean? Can you provide link or example?

Reply:
Once upon a time It was a tradition to use original plugin's author code style and naming conventions in PR despite the fact that sometimes they are inconsistent, e.g. you can find in code:
https://github.com/interiorem/cocaine-plugins/blob/7c5df8ff03ac9bd0e182a525eac19dc4ddc988a8/vicodyn/src/vicodyn/proxy.cpp#L34 

So it could be `app_predicate_t` in your case, etc.

Reply:
Thank you. Fixed.